test/e2e/common.go

```diff
@@ -47,6 +47,30 @@ func createNamespace(ctx context.Context, client testClient, namespace string) e
 	return err
 }
 
+// createNamespaceAndAnnotation creates a kubernetes namespace and an annotation
+func createNamespaceAndAnnotation(ctx context.Context, client testClient, namespace string, annotation string) error {
+	ns := builder.ForNamespace(namespace).Result()
+	_, err := client.clientGo.CoreV1().Namespaces().Create(ctx, ns, metav1.CreateOptions{})
+
+	if apierrors.IsAlreadyExists(err) {
+		return nil
+	}
+
+	nsAnno := ns.ObjectMeta.Annotations
+
+	if nsAnno == nil {
+		nsAnno = make(map[string]string)
+	}
+	nsAnno["testAnnotation"] = annotation
+	ns.ObjectMeta.Annotations = nsAnno
+	_, err = client.clientGo.CoreV1().Namespaces().Update(ctx, ns, metav1.UpdateOptions{})
+
+	if apierrors.IsAlreadyExists(err) {
+		return nil
+	}
+	return err
```

Comment:
This is an Update call, so I wouldn't expect an IsAlreadyExists error. Just return err at this point.

Reply:
added changes in new commit